### PR TITLE
feat: incremental sync update user events (AR-2491)

### PR DESCRIPTION
**Pull request by yamilmedina**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Incremental sync support, handling user events.
With this PR we are handling user events, related to when they change personal information, so we act on this, and update the target local user, ie: username, handle, email, avatars.

### Testing

 WIP

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

You can check out this branch and use it in reloaded, you will see other users info changed automatically.
I´ll open a PR on reloaded for this updated handling

### Notes (Optional)

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by Garzas — APPROVED** [review]
Nice one 🎉

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserRepository.kt [hunk 1]
```diff
@@ -291,6 +295,12 @@ internal class UserDataSource internal constructor(
             }
             .map(memberMapper::fromMapOfClientsToRecipients)
 
+    override suspend fun updateUserFromEvent(event: Event.User.Update): Either<CoreFailure, Unit> = wrapStorageRequest {
+        val userId = qualifiedIdMapper.fromStringToQualifiedID(event.userId!!)
```
Comment (typfel) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserRepository.kt:299]: We shouldn't force unwrap the userId, but is the `userId` really optional on the user.update event?

**Review by typfel — CHANGES_REQUESTED** [review]

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserRepository.kt [hunk 2]
```diff
@@ -291,6 +295,12 @@ internal class UserDataSource internal constructor(
             }
             .map(memberMapper::fromMapOfClientsToRecipients)
 
+    override suspend fun updateUserFromEvent(event: Event.User.Update): Either<CoreFailure, Unit> = wrapStorageRequest {
+        val userId = qualifiedIdMapper.fromStringToQualifiedID(event.userId)
+        val user = userDAO.getUserByQualifiedID(idMapper.toDaoModel(userId)).firstOrNull()
+        userDAO.updateUser(userMapper.toUpdateDaoFromEvent(event, user!!))
```
Comment (typfel) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserRepository.kt:301]:
```suggestion
        userDAO.getUserByQualifiedID(idMapper.toDaoModel(userId)).firstOrNull()?.let { user ->
                  userDAO.updateUser(userMapper.toUpdateDaoFromEvent(event, user))
        }
```

Comment (typfel) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserRepository.kt:301]: Or maybe return Either.Left if the user is missing.

Comment (typfel) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/user/UserRepository.kt:301]:
```suggestion
        val user = userDAO.getUserByQualifiedID(idMapper.toDaoModel(userId)).firstOrNull() ?: return Either.Left(StorageFailure.DataNotFound)
        userDAO.updateUser(userMapper.toUpdateDaoFromEvent(event, user))
```

**Review by typfel — APPROVED** [review]

**Pull request merged**